invokeai/app/invocations/facetools.py

```diff
@@ -0,0 +1,654 @@
+import math
+
+import cv2
+import mediapipe as mp
+import numpy as np
+from PIL import Image, ImageDraw, ImageFilter, ImageOps
+
+from invokeai.app.invocations.baseinvocation import (
+    BaseInvocation,
+    BaseInvocationOutput,
+    InputField,
+    InvocationContext,
+    OutputField,
+    invocation,
+    invocation_output,
+)
+from invokeai.app.invocations.primitives import ImageField, ImageOutput
+from invokeai.app.models.image import ImageCategory, ResourceOrigin
+
+
+@invocation_output("face_mask_output")
+class FaceMaskOutput(BaseInvocationOutput):
+    """Base class for FaceMask output"""
+
+    image: ImageField = OutputField(description="The output image")
+    width: int = OutputField(description="The width of the image in pixels")
+    height: int = OutputField(description="The height of the image in pixels")
+    mask: ImageField = OutputField(description="The output mask")
```

Comment:
This can inherit from `ImageOutput` and automatically get the `image`, `width` and `height` fields. Unless you want to do something different, this is preferrable to creating a new output class.

```suggestion
@invocation_output("face_mask_output")
class FaceMaskOutput(ImageOutput):
    """Base class for FaceMask output"""

    mask: ImageField = OutputField(description="The output mask")
```